### PR TITLE
Station Anchor

### PR DESCRIPTION
**Pull request by juliangiebel**
## About the PR
This PR adds the station anchor. A machine that prevents grids from moving if active.
There is a version that doesn't use power and is indestructible 

This PR also abstracts out the charging functionality and ui of the gravity generator into the `PowerChargeSystem`

## Why / Balance

Putting this on the trade station prevents players from moving the trade station

## Technical details

Sets grids to static when active

## Media

image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## TODO:
Setting the grid to static prevents it from showing up on the shuttle console and it prevents the arrivals shuttle from docking.

**Changelog**

:cl:
- add: Adds the station anchor. It anchors stations in space and prevents them from moving.

### CODE REVIEW

**Comment by UbaserB** [comment]
> I thought about this solution for players moving trade station to the main station but... They can still move **_the main station to the trade station_**. And we don't want to anchor the main station.
> 
> I came up with the idea that the relative position between trade station and main station should be maintained (ideally via a soft tether). The lore explanation is that main station + trade station use the joint center mass to maintain orbit, which is why they are artificially kept an equal vector from each other.
> 
> I mean this is fine as a feature and works as a stopgap solution (it's more difficult to move the main station to the trade station than the other way around) but the underlying problem remains unsolved.

station anchor is being worked on lol

**Comment by UbaserB** [comment]
> > station anchor is being worked on
> 
> I don't understand. Can you elaborate? What station anchor? The one in this draft? I can see that so I'm commenting on it. If you mean another one then please point me in the right direction.

why did i think this was a comment for another pr

**Comment by juliangiebel** [comment]
The main station should also be anchored.
Also please move discussions about this to discord or github discussions

**Comment by chromiumboy** [comment]
The sprite looks really cool, but is there reason why the anchor is separate from the gravity generator? Anchoring seems like a reasonable side effect of gravity manipulation, and it would save mappers having to include another structure (that needs to be protected with restricted access?)

Also, making the grid static seems to be causing you some issues, based on your todo list. You can't just increase the mass of the grid to some ludicrously high number, or change its max velocity, rate of acceleration, or friction coefficient, to dampen movement to be effectively zero?

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Comment by EmoGarbage404** [comment]
8373905e82c8bf5c0a093f984bf135e5

**Comment by EmoGarbage404** [comment]
The shuttle console and docking issues appear to be fixed. @juliangiebel let me know if there were other bugs that you encountered. Otherwise, this is ready for review.

File: Content.Server/Power/Components/PowerChargeComponent.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using Content.Server.Power.EntitySystems;
+using Content.Shared.Power;
+
+namespace Content.Server.Power.Components;
+
+[RegisterComponent]
+[Access(typeof(PowerChargeSystem))]
+public sealed partial class PowerChargeComponent : SharedPowerChargeComponent
+{
+    // TODO: Change comments
```
Comment (PJB3005) [inline Content.Server/Power/Components/PowerChargeComponent.cs:10]: Nerd

**Comment by PJB3005** [comment]
Test failure is legit 
image

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**